kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/ClientConsumerRecord.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+
+/**
+ * The type Client consumer record.
+ */
+public class ClientConsumerRecord extends BaseConsumerRecord {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientConsumerRecord.class);
+
+    private long timestamp;
+    private String timestampType;
+    private List<Map<String, String>> headers;
+
+    /**
+     * Sets timestamp.
+     *
+     * @param timestamp the timestamp
+     */
+    public void setTimestamp(long timestamp) {
+        this.timestamp = timestamp;
+    }
+
+    /**
+     * Sets timestamp type.
+     *
+     * @param timestampType the timestamp type
+     */
+    public void setTimestampType(String timestampType) {
+        this.timestampType = timestampType;
+    }
+
+    /**
+     * Sets headers.
+     *
+     * @param headers the headers
+     */
+    public void setHeaders(List<Map<String, String>> headers) {
+        this.headers = headers;
+    }
+
+    /**
+     * Parse from json string.
+     *
+     * @param response the response
+     * @return the client consumer record
+     */
+    public static ClientConsumerRecord parseFromJsonString(String response) {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        try {
+            return objectMapper.readValue(response, ClientConsumerRecord.class);
```

Comment:
Jackson can handle deserializing a stream of json objects to a List.  You don't need to deserialise each one individually.